magik-checks/src/main/resources/nl/ramsolutions/sw/sonar/l10n/magik/rules/NestingDepth.html

```diff
@@ -0,0 +1,34 @@
+<p>The nesting depth of methods/procedures/if-statements/loop-statements should not exceed the defined maximum nesting depth threshold. Deeply nested code may perform poorly, is difficult to understand and can be difficult to test thoroughly.</p>
+<h2>Noncompliant Code Example</h2>
+<pre>
+_if a _isnt _unset
+_then
+  _for x _over a.fast_elements()
+  _loop
+    _if x _isnt _unset
+    _then
+      _for child _over x.fast_elements()
+      _loop
+        _return child
+      _endloop
+    _endif
+  _endloop
+_else
+  _return b
+_endif
+</pre>
+<h2>Compliant Solution</h2>
+<pre>
+_if a _is _unset _then _return b _endif
+
+_for x _over a.fast_elements()
+_loop
+  _if x _isnt _unset
+  _then
```

Comment:
You could further improve (subjective, of course) that you can rewrite this as:
```
_for x _over a.fast_elements()
_loop
  _if x _is _unset
  _then
    _continue
  _endif

  _for child _over x.fast_elements()
  _loop
    _return child
  _endloop
_endloop
```

Or even write the `_if x ...` on a single line. Though for clarity I wouldn't do this.